### PR TITLE
Testing environment variables (v3)

### PR DESCRIPTION
**Pull request by acesnik**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by acesnik** [comment]
Added testing of DOTNET_ROOT and DOTNET_TOOLS. Those are ones I've used. Should we test all the ones set in `build.sh`?

**Comment by dhirschfeld** [comment]
Sorry for the radio silence - I'm spread a bit too thin to keep up with things 😬 

> *Should we test all the ones set in `build.sh`?*

I think we should probably aim towards that goal, but we don't need to get there in a single PR. Any individual improvement in that direction can (and should IMHO) be merged - perfect is the enemy of shipped as they say!

Not sure what's going on with OSX here (and I don't have access to OSX to test). I'll have a poke about in the next few weeks to try and figure it out if you don't beat me to it...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -58,6 +58,10 @@ outputs:
     test:
       commands:
         - dotnet --list-runtimes
+        - (! test -z $DOTNET_ROOT)            # [unix]
+        - (! test -z $DOTNET_TOOLS)           # [unix]
```
Comment (dhirschfeld) [inline recipe/meta.yaml:62]:
```suggestion
        - (! test -z "${DOTNET_ROOT}")            # [unix]
        - (! test -z "${DOTNET_TOOLS}")           # [unix]
```

**Comment by dhirschfeld** [comment]
☝️ thought I'd give one thing a go. Unfortunately, my bash skills are ~~rubbish~~still a work in progress...

**Comment by dhirschfeld** [comment]
Looks like the same error still :/
```
+ test -z /Users/runner/miniforge3/conda-bld/dotnet_1638267189740/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/dotnet
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -58,6 +58,10 @@ outputs:
     test:
       commands:
         - dotnet --list-runtimes
+        - (! test -z "${DOTNET_ROOT}")            # [unix]
+        - (! test -z "${DOTNET_TOOLS}")           # [unix]
```
Comment (dhirschfeld) [inline recipe/meta.yaml:62]:
```suggestion
        - test -z "${DOTNET_ROOT}" = "${CONDA_PREFIX}/lib/dotnet"         # [unix]
        - test -z "${DOTNET_TOOLS}" = "${CONDA_PREFIX}/lib/dotnet/tools"  # [unix]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -58,6 +58,10 @@ outputs:
     test:
       commands:
         - dotnet --list-runtimes
+        - test -z "${DOTNET_ROOT}" = "${CONDA_PREFIX}/lib/dotnet"         # [unix]
+        - test -z "${DOTNET_TOOLS}" = "${CONDA_PREFIX}/lib/dotnet/tools"  # [unix]
```
Comment (dhirschfeld) [inline recipe/meta.yaml:62]:
```suggestion
        - test "${DOTNET_ROOT}" = "${CONDA_PREFIX}/lib/dotnet"         # [unix]
        - test "${DOTNET_TOOLS}" = "${CONDA_PREFIX}/lib/dotnet/tools"  # [unix]
```

**Review by dhirschfeld — APPROVED** [review]
Not sure why the original version didn't work (on OSX) but since this accomplishes the ~same thing I suggest we merge and iterate if you're happy with the changes @acesnik?

:shipit:

**Comment by acesnik** [comment]
Looks good for now!

**Pull request merged**